owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt

```diff
@@ -367,16 +368,14 @@ class MainFileListFragment : Fragment(),
      * @param newFileListOption new file list option to enable.
      */
     private fun showOrHideFab(newFileListOption: FileListOption, currentFolder: OCFile) {
-        if (!newFileListOption.isAllFiles() || isPickingAFolder() || (currentFolder.permissions != null && !currentFolder.hasAddFilePermission && !currentFolder.hasAddSubdirectoriesPermission)) {
+        if (!newFileListOption.isAllFiles() || isPickingAFolder() || (!currentFolder.hasAddFilePermission && !currentFolder.hasAddSubdirectoriesPermission)) {
             toggleFabVisibility(false)
         } else {
             toggleFabVisibility(true)
-            if (currentFolder.permissions != null) {
-                if (!currentFolder.hasAddFilePermission) {
-                    binding.fabUpload.isVisible = false
-                } else if (!currentFolder.hasAddSubdirectoriesPermission) {
-                    binding.fabMkdir.isVisible = false
-                }
+            if (!currentFolder.hasAddFilePermission) {
+                binding.fabUpload.isVisible = false
+            } else if (!currentFolder.hasAddSubdirectoriesPermission) {
+                binding.fabMkdir.isVisible = false
```

Comment:
```suggestion
            binding.fabUpload.isVisible = currentFolder.hasAddFilePermission
            binding.fabMkdir.isVisible = currentFolder.hasAddSubdirectoriesPermission
            binding.fabMain.isVisible = current.Folder.hasAddFilePermission || currentFolder.hasAddSubdirectoriesPermission
```

Reply:
That's not the same logic! Now only one of the two mini FABs is hidden here, since if the folder has none of the two permissions, it will enter in the previous `if`, which is also in charge for hiding the main FAB with `toggleFabVisibility(false)`